### PR TITLE
add python solution for modern art (737)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-737.mdx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+<PySection>
+
+```py
+res = [[0] * 11 for _ in range(11)]
+left, right, down, up = [0] * 10, [0] * 10, [0] * 10, [0] * 10
+
+ans = [0] * 10
+
+for i in range(1, 10):
+        left[i], up[i] = 1e9, 1e9
```
Comment (SansPapyrus683) [inline solutions/usaco-737.mdx:24]: use `float('inf')` for effectively infinite values

File: solutions/usaco-737.mdx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Implementation
+**Time Complexity**: $\mathcal O(N^3)$
+
+<LanguageSection>
+
+<PySection>
+
+```py
+res = [[0] * 11 for _ in range(11)]
```
Comment (SansPapyrus683) [inline solutions/usaco-737.mdx:18]: magic number
put 10 in a variable like WIDTH or something (all caps)

File: solutions/usaco-737.mdx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+left, right, down, up = [0] * 10, [0] * 10, [0] * 10, [0] * 10
+
+ans = [0] * 10
```
Comment (SansPapyrus683) [inline solutions/usaco-737.mdx:21]: better names than `res` & `ans`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-737.mdx (hunk 3 above)
Comment (maggie-j-liu) [reply solutions/usaco-737.mdx:21]: wait i think these names are fine 👀

File: solutions/usaco-737.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-737.mdx:18]: okay.

File: solutions/usaco-737.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-737.mdx]: ill change it if u give me suggestion

Comment (SansPapyrus683) [reply solutions/usaco-737.mdx]: what do they store? what does it mean in the context of the problem?

Comment (jjliewie) [reply solutions/usaco-737.mdx]: ans stores basically 1 if the possible number can be the "starting number" -> the question is asking how many possible starting numbers 

and res stores the "art" or the input

File: solutions/usaco-737.mdx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+**Time Complexity**: $\mathcal O(N^3)$
+
+<LanguageSection>
+
+<PySection>
+
+```py
+WIDTH = 10
+
+res = [[0] * (WIDTH + 1) for _ in range(WIDTH + 1)]
```
Comment (SansPapyrus683) [inline solutions/usaco-737.mdx:20]: rename to `art`

File: solutions/usaco-737.mdx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+left, right, down, up = [0] * WIDTH, [0] * WIDTH, [0] * WIDTH, [0] * WIDTH
+
+ans = [0] * WIDTH
```
Comment (SansPapyrus683) [inline solutions/usaco-737.mdx:23]: 1. rename to `valid_start`
2. use bools instead of ints

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-737.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-737.mdx:23]: yeah but then i cant sum it

Comment (SansPapyrus683) [reply solutions/usaco-737.mdx]: i just ran this on a mac
```py
Python 3.8.9 (default, Jul 19 2021, 09:37:32) 
[Clang 13.0.0 (clang-1300.0.27.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = [True, False]
>>> sum(x)
1
>>> 
```

Comment (jjliewie) [reply solutions/usaco-737.mdx]: oh nice i'll change it then

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
⭐

**Pull request merged**